### DISCUSSION TITLE
Why Stream Queue's can not consume messages on the last offset when restart consumer?

### DISCUSSION THREAD

**Discussion by LucasZhanye** [rabbitmq/amqp091-go#278] [Q&A] [answered]
I have a problem about stream queue

I have declare a consumer with property "x-stream-offset": "next", when I consume message, It is OK!

And then I close the consumer,and producer still send message to the queue.

At Last I start consumer again, But I can't keep consuming since the last offset

I don't know what the "next" mean?

there is my code:

```
err = channel2.Qos(10, 0, false)
	msgs2, err := channel2.Consume(
		"lucas.products",
		"C12",
		false,
		false,
		false,
		false,
		amqp.Table{
			"x-stream-offset": "next",
		},
	)
	if err != nil {
		log.Fatal(err, "Failed to register a consumer")
	}
	go func() {
		for d := range msgs2 {
			fmt.Printf("customer:%s\n", d.Body)
			if err := d.Ack(false); err != nil {
				log.Printf("Failed to acknowledge message: %s", err)
			} else {
				log.Printf("Acknowledged message")
			}
		}
	}()

	<-forever
```

**Comment by Zerpet** [comment, marked as answer]
You have the meaning for each offset specification documented here: https://www.rabbitmq.com/docs/streams#client-operations

You have to keep track of the last offset that your consumer acknowledges. If you want to start consuming from the last ack'ed offset, you have to set the stream offset as `offset` and specify the last consumed offset + 1. If you want to use automatic consumer tracking, you have to enable the stream plugin, and use one of the stream clients: https://www.rabbitmq.com/docs/stream

**Reply by LucasZhanye** [reply]
Thanks!!
